torch/onnx/_internal/exporter/_torchlib/ops/hop.py

```diff
@@ -0,0 +1,90 @@
+"""Implementation for higher-order operators."""
+
+from __future__ import annotations
+
+from typing import Sequence
+
+import torch
+from torch.onnx._internal._lazy_import import onnxscript_ir as ir
+from torch.onnx._internal.exporter import _core
+from torch.onnx._internal.exporter._torchlib._torchlib_registry import onnx_impl
+
+
+def call_op(
+    op_type: str,
+    *args: ir.Value,
+    _num_outputs: int = 1,
+    _domain: str = "",
+    **kwargs: int | float | str | bool | ir.Graph | ir.TensorProtocol,
+) -> Sequence[ir.Value]:
+    """Call an operator with the given arguments and keyword arguments.
+
+    Arguments are always inputs, while keyword arguments are attributes.
+    """
+    # This is a wrapper around the IR node creation that hooks into the _builder.OpRecorder
+    # tracer so that all nodes created are recorded the same way as if we were to use
+    # onnxscript ops directly.
+    from onnxscript.ir import convenience as ir_convenience
+
+    assert _core.current_tracer is not None
+    tracer = _core.current_tracer
```

Comment:
This will go away when torchlib is moved to pytorch?

Reply:
Not necessarily? We need to find a more elegant way to capture the node creation but it should be independent of the torchlib migration. Do you have some ideas?

Reply:
No, I was confused by the extra lines about tracer because I thought we were still using eval and eval_function that are defined in tracer.

Reply:
We are just intercepting the tracer and writing to its `nodes` property directly. This tracer is owned by the exporter so we can do anything to it.

Reply:
Does it mean we are introducing a function that are not ONNXFunction and TracedONNXFunction?

Reply:
Yes. This is a plain function that has full access to the graph (more flexibility for HOPs), but it can be limited to HOPs for now